src/Microsoft.ML.DataView/DataViewRowCursor.md

```diff
@@ -0,0 +1,218 @@
+﻿# `DataViewRowCursor` Notes
+
+This document includes some more in depth notes on some expert topics for
+`DataViewRow` and `DataViewRowCursor` derived classes.
+
+## `Batch`
+
+The `Batch` property is the mechanims by which the data from the multiple
+cursors returned by `IDataView.GetRowCursorSet` can be reconciled into a
+single, cohesive, sequence.
+
+`IDataView.GetRowCursorSet` can return a set of parallel cursors that
+partition the sequence of rows as would have normally been returned through a
+more typical call to `GetRowCursor`. These cursors can be accessed across
+multiple threads to enable parallel evaluation of a data pipeline. This is key
+for the data pipeline performance during featurization.
+
+However, even though the data pipeline can perform this parallel evaluation,
+at the end of this parallelization we sometimes want to recombine the separate
+thread's streams back into a single cohesive sequence. This can be done for
+several reasons: we may want repeatability and determinism, or the cursor may
+be stateful in some way that precludes the sharding inherent . But, for this end consumer that calls
+the single cursor. And, since a core `IDataView` design principle is
+repeatability, we now have a problem of how to reconcile those separate
+partitioninging. This reconciliation to re-sequence the partitioned dataset is
+accomplished through `DataViewRow.Batch`.
+
+So, to review what actually happens in this ML.NET utility code: multiple
```

Comment:
>So, to revie [](start = 0, length = 12)

I might start with this paragraph. Right now, you have introduction => preamble => let me explain. If you have time, it might help to start with the big picture and then give examples, rather than start with an example.

Reply:
OK... Both you and @sfilipi gave the same feedback, I am not sure I like how it flows, but, if two people tell me the same exact thing I start to suspect there's something I'm missing, so I'll try it out.

Reply:
All right -- the description is a bit tighter, but hopefully OK? I hope? Anyway if not it can be improved later. :) Or not.